coverage/data.py

```diff
@@ -70,7 +70,7 @@ def combinable_files(data_file, data_paths=None):
         if os.path.isfile(p):
             files_to_combine.append(os.path.abspath(p))
         elif os.path.isdir(p):
-            pattern = os.path.join(os.path.abspath(p), f"{local}.*")
```

Comment:
The `p` here should also be escaped, there is no guarantee that it won't have `[` or `]` in it